src/CachePlugin.php

```diff
@@ -36,11 +42,13 @@ class CachePlugin implements Plugin
 
     /**
      * @param CacheItemPoolInterface $pool
+     * @param StreamFactory          $streamFactory
      * @param array                  $options
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $options = [])
```

Comment:
Yes, you would miss the validation from the option resolver. But you would not have the issue with default values and misstyped array keys.